src/sandstorm/run-bundle.c++

```diff
@@ -1162,12 +1162,23 @@ private:
 
     // The environment inherited from the host is probably no good for us. E.g. an oddball
     // locale setting can crash Mongo because we don't have the appropriate locale files available.
+    //
+    // That said, there are a few environment variables that we do re-export.
+    char * http_proxy, * https_proxy;
+    KJ_SYSCALL(http_proxy = getenv("https_proxy"));
+    KJ_SYSCALL(https_proxy = getenv("https_proxy"));
     KJ_SYSCALL(clearenv());
```

Comment:
I believe that this `clearenv()` call frees the memory that your `http_proxy` and `https_proxy` pointers are pointing at. You will need to copy the strings, preferably using `kj::str()` -- but of course only if they aren't null.